internal/expr/expr_test.go

```diff
@@ -307,6 +311,24 @@ AND z = @sqlair_0 -- The line with $Person.id on it
 	"[Bypass[INSERT INTO arr VALUES (ARRAY[[1,2],[] Input[HardMaths.x] Bypass[,4]], ARRAY[[5,6],[] Input[HardMaths.y] Bypass[,8]]);]]",
 	[]any{HardMaths{}},
 	"INSERT INTO arr VALUES (ARRAY[[1,2],[@sqlair_0,4]], ARRAY[[5,6],[@sqlair_1,8]]);",
+}, {
+	"all primitive types",
+	"SELECT (col1, col2, col3, col4, col5, col6, col7, col8, col9, col10, col11, col12, col13, col14) AS (&string, &bool, &uint, &uint8, &uint16, &uint32, &uint64, &int, &int8, &int16, &int32, &int64, &float32, &float64) FROM t",
```

Comment:
This test is actually one that clearly demonstrates my concern.
What is the type of `col8`, how long does it take you to figure out whether it is actually a correct statement, and that you are actually parsing the right value into the correct field.

If you even only did:
```
SELECT (col1 AS &string), (col2 AS &bool)...
```
you would have much more locality and a better understanding of whether the statement is actually correct. Put another way, can you tell that this statement is wrong:
```
"SELECT (col1, col2, col3, col4, col5, col6, col7, col8, col9, col10, col11, col12, col13, col14) AS (&string, &bool, &uint, &uint, &uint8, &uint16, &uint32, &uint64, &int, &int8, &int16, &int32, &int64, &float32, &float64) FROM t"
```

Reply:
That's very fair, I had written this in the less readable form to make the test more concise. You make a very good point that people will use these tests to get a feel for how to use the new constructs. I have changed it to the other form.